juju/jasyncio.py

```diff
@@ -82,6 +83,13 @@
 
 import websockets
 
+warnings.warn(
+    "juju.jasyncio module is being deprecated by 3.0, use asyncio or juju._jasyncio instead",
+    DeprecationWarning,
+    stacklevel=2,
+)
+
+
```

Comment:
I think this file should re-export the useful _jasyncio functions instead of having own copies.

Only the `run` function should remain defined in this file.

(asyncio re-exports are fine).